From c7cecdaa0a582225f94c24ce7e250bdb59c61c39 Mon Sep 17 00:00:00 2001
From: Mohammed Sameer <msameer@foolab.org>
Date: Mon, 14 Apr 2014 22:23:11 +0300
Subject: [PATCH 4/6] Set video branch to NULL after finishing video recording.

This patch has been extracted rom Sailfish OS public source code release
and ported to GStreamer 1.x
Original patch: 0003-Set-video-branch-to-NULL-after-finishing-video-recor.patch
---
 gst/camerabin2/gstcamerabin2.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gst/camerabin2/gstcamerabin2.c b/gst/camerabin2/gstcamerabin2.c
index a73e175..8638859 100644
--- a/gst/camerabin2/gstcamerabin2.c
+++ b/gst/camerabin2/gstcamerabin2.c
@@ -976,10 +976,10 @@ gst_camera_bin_video_reset_elements (gpointer u_data)
   gst_camera_bin_finish_video_file (camerabin);
 
   /* reset element states to clear eos/flushing pads */
-  gst_element_set_state (camerabin->video_encodebin, GST_STATE_READY);
-  gst_element_set_state (camerabin->videobin_capsfilter, GST_STATE_READY);
+  gst_element_set_state (camerabin->video_encodebin, GST_STATE_NULL);
+  gst_element_set_state (camerabin->videobin_capsfilter, GST_STATE_NULL);
   if (camerabin->video_filter) {
-    gst_element_set_state (camerabin->video_filter, GST_STATE_READY);
+    gst_element_set_state (camerabin->video_filter, GST_STATE_NULL);
     gst_element_sync_state_with_parent (camerabin->video_filter);
   }
   gst_element_sync_state_with_parent (camerabin->videobin_capsfilter);
-- 
1.8.5.2

